src/main/java/newbank/server/NewBank.java

```diff
@@ -42,10 +45,24 @@ private void addTestData() {
    * @param username The customer's username
    * @param password The customer's password
    */
-  public void addCustomer(final String username, final String password) {
+  public void addCustomer(final String username, final String password)
+      throws DuplicateCustomerException, PasswordInvalidException {
+    if (customers.containsKey(username)) {
+      throw new DuplicateCustomerException();
+    }
+
+    validatePassword(password);
+
     customers.put(username, new Customer(username, password));
   }
 
+  // Simple algorithm to check that the password meets the security requirements
+  private void validatePassword(final String password) throws PasswordInvalidException {
+    if (password.isEmpty()) {
+      throw new PasswordInvalidException();
```

Comment:
Possibly could be `CustomerPasswordInvalidException` - unless we are considering a general notion of a password. But again - I do realize I'm bikeshedding a lil bit :))

Reply:
This is will vanish in my next PR.